docs/explanation/searchdocs.rst

```diff
@@ -1,2 +1,131 @@
 Search and django CMS
 =====================
+
+.. seealso::
+
+    - :ref:`Version States <version_states>`
+    - `djangocms-versioning <https://github.com/django-cms/djangocms-versioning>`_
+    - `django-haystack <https://github.com/django-haystack/django-haystack>`_
+
+If you already have a background in django CMS and are coming from the 3.x days, you may 
+be familiar with the package ``aldryn-search`` which was the recommended way for setting up
+search inside your django CMS project. However, with the deprecation of it and the new
+versioning primitives in django CMS 4.x being incompatible with it, you may wonder where to
+go from here. 
+
+To keep it simple, you can install an external package that handles implementing the search 
+index for you. There are multiple options, such as
+
+* `djangocms-aldryn-search <https://github.com/CZ-NIC/djangocms-aldryn-search>`_
+* `djangocms-haystack <https://github.com/Lfd4/djangocms-haystack>`_
+* ... and others
+
+The idea is that we implement the search index just as we would for normal Django models,
+e.g. use an external library to handle the indexing (``django-haystack`` in the above examples)
+and just implement the logic that builds the querysets and filters the languages depending on
+what index is currently active. ``django-haystack`` exposes many helpful utilities for 
+interacting with the indexes and return the appropriate results for indexing.
+
+To get an idea on how this works, feel free to look at the code of the above projects.
+A very simple index could look something like this:
+
+..  code-block:: python
+    :emphasize-lines: 3
+
+    from cms.models import PageContent
+    from django.db import models
+    from haystack import indexes
+
+
+    class PageContentIndex(indexes.SearchIndex, indexes.Indexable):
+        text = indexes.CharField(document=True, use_template=False)
+        title = indexes.CharField(indexed=False, stored=True)
+        url = indexes.CharField(indexed=False, stored=True)
+
+        def get_model(self):
+            return PageContent
+
+        def index_queryset(self, using=None) -> models.QuerySet:
+            return self.get_model().objects.filter(
+              versions__state="published",
```

Comment:
I don't think the `versions_state` filter is necessary (indeed, maybe confusing, since you might not have installed djangocms-versioning). djangocms-versioning or any other versioning package must modify the `objects` manager so that it only returns published objects anyway.

Reply:
Good catch, I didn't know this yet. Your proposed change is committed.

Reply:
@filipweidemann Could you also test it?

Reply:
Oh I didn't see this, I did test this & merged your PR in our own version earlier today. Works as intended ;)
https://github.com/Lfd4/djangocms-haystack/pull/12#issuecomment-2258246830